### ISSUE TITLE
False positive for unsupported-membership-test in custom data structure

### ISSUE THREAD

**Issue by LWprogramming** [PyCQA/pylint#3743]
### Steps to reproduce
```
# pylint: disable=missing-docstring, invalid-name
from graphql.pyutils import FrozenDict

f = FrozenDict(
    {
        'a': 1,
        'b': 2
    }
)

if 'a' in f:
    print('a in f')
```

### Current behavior
```
************* Module test
test.py:11:10: E1135: Value 'f' doesn't support membership test (unsupported-membership-test)
```

### Expected behavior
```
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### pylint --version output
```
pylint 2.4.4
astroid 2.3.3
Python 3.8.0 (default, Oct 28 2019, 16:14:01) 
[GCC 8.3.0]
```

`graphql-core` version is 3.1.2

Note that `FrozenDict` [inherits](https://github.com/graphql-python/graphql-core/blob/6687b168b82e6c1722bb9748430f8785cabe46f5/src/graphql/pyutils/frozen_dict.py#L12) from `Dict`, but `pylint` does not complain when `f` is a dict instead of a `FrozenDict`.

----

Update: similar problem if, instead of membership, we try `if a in f.keys()`. That raises `test.py:11:10: E1101: Instance of 'FrozenDict' has no 'keys' member (no-member)`

**Comment by AWhetter** [comment]
I'm going to close this in favour of #2420 because it covers a wider set of issues associated with inheriting from `typing.Dict`.
